base_layer/core/src/proof_of_work/randomx_factory.rs

```diff
@@ -9,13 +9,13 @@ use std::{
 
 const LOG_TARGET: &str = "c::pow::randomx_factory";
 
-#[derive(Default)]
+#[derive(Default, Clone, Copy)]
 pub struct RandomXConfig {
     pub use_large_pages: bool,
 }
 
-impl From<&RandomXConfig> for RandomXFlag {
-    fn from(source: &RandomXConfig) -> Self {
+impl From<RandomXConfig> for RandomXFlag {
+    fn from(source: RandomXConfig) -> Self {
         let mut result = RandomXFlag::get_recommended_flags();
         if source.use_large_pages {
```

Comment:
I think `source.use_large_pages` here should be removed. Useful for mining on supported machines, not for validation.

Reply:
Yip, it seems that if `RandomXConfig` is not exposed in the config file and can only either be recommended or default (as implemented), then we do not need that struct at all,

Reply:
Inclined to agree, removed